perllib/FixMyStreet/Cobrand/UK.pm

```diff
@@ -336,5 +336,20 @@ sub report_check_for_errors {
     return %errors;
 }
 
+sub get_body_handler_for_problem {
+    my ($self, $row) = @_;
+
+    foreach my $body (split(/,/, $row->bodies_str)) {
+        foreach my $avail ( FixMyStreet::Cobrand->available_cobrand_classes ) {
+            my $class = FixMyStreet::Cobrand->get_class_for_moniker($avail->{moniker});
+            my $cobrand = $class->new( ref $self ? { %$self } : {} );
+            next unless $cobrand->can('council_id');
+            # $c->log->debug( sprintf "Set body to '%s' and cobrand to '%s', which has council id %s", $body, $cobrand->moniker, $cobrand->council_id );
+            return $cobrand if int($body) eq $cobrand->council_id;
```

Comment:
I've refactored this to build a hash of cobrands by ID, but don't think I'm quite sure what you mean here.
Are you suggesting using Problem's `areas` instead of `bodies_str` for determining which cobrand to link to? The issue I found there was that the parent council's area ID is included in `areas` even if the report was sent to the district council, so you end up with a link to view the problem on the parent council's cobrand.